polus-basic-flatfield-correction-plugin/src/basic.py

```diff
@@ -224,47 +221,47 @@ def _inexact_alm_l1(D,options):
             B1_offset = np.max(B1_offset,initial=0)
             B1_offset = np.min(B1_offset,initial=B1_uplimit/(np.mean(W_idct_hat)+10**-7))
             B_offset = B1_offset * np.mean(W_idct_hat) - B1_offset*np.reshape(W_idct_hat,(-1,1))
-            
+
             # Calculate darkfield
             A1_offset = np.reshape(np.mean(R1[:,validA1coeff_idx],1),(-1,1)) - np.mean(A_coeff[0,validA1coeff_idx]) * np.reshape(W_idct_hat,(-1,1))
             A1_offset = A1_offset - np.mean(A1_offset)
             A_offset = A1_offset - np.mean(A1_offset) - B_offset
-            
+
             # Update darkfield weights
-            W_offset = _dct2(np.reshape(A_offset,(p,q)))
-            W_offset = np.max(np.reshape(W_offset - options['lambda_darkfield']/(ent2*mu),(p,q,1)),-1,initial=0) \
-                     + np.min(np.reshape(W_offset + options['lambda_darkfield']/(ent2*mu),(p,q,1)),-1,initial=0)
+            W_offset = _dct2(np.reshape(A_offset,(img_width, img_height)))
+            W_offset = np.max(np.reshape(W_offset - options['lambda_darkfield']/(ent2*mu),(img_width, img_height,1)),-1,initial=0) \
+                     + np.min(np.reshape(W_offset + options['lambda_darkfield']/(ent2*mu),(img_width, img_height,1)),-1,initial=0)
 
             # Calculate darkfield based on updated weights
             A_offset = _idct2(W_offset)
             A_offset = np.reshape(A_offset,(-1,1))
             A_offset = np.max(np.reshape(A_offset - options['lambda_darkfield']/(ent2*mu),(A_offset.shape[0],A_offset.shape[1],1)),-1,initial=0) \
                      + np.min(np.reshape(A_offset + options['lambda_darkfield']/(ent2*mu),(A_offset.shape[0],A_offset.shape[1],1)),-1,initial=0)
             A_offset = A_offset + B_offset
-            
+
         # Loss
-        Z1 = D - A - E1
-        
+        Z1 = imgf_stack - A - E1
+
         # Update weight regularization term
         Y1 = Y1 + mu*Z1
-        
+
         # Update learning rate
         mu = np.min(mu*rho,initial=mu_bar)
-        
+
         # Stop if loss is below threshold
         stopCriterion = np.linalg.norm(Z1,ord='fro')/d_norm
         if stopCriterion < options['optimization_tol'] or iternum > options['max_iterations']:
             converged = True
-        
+
     # Calculate final darkfield image
     A_offset = A_offset + B1_offset * np.reshape(W_idct_hat,(-1,1))
-    
+
     return A,E1,A_offset
 
 
 def _get_flatfield_and_reweight(X_k_A,X_k_E,X_k_Aoffset,options):
```

Comment:
The input values should be renamed to something more human readable. Also, I notice that I messed up the names of the input values in the description.